lib/cluster/workers/IDCardRenewer.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+const { isMainThread, parentPort } = require('worker_threads');
+const Redis = require('../../service/cache/redis');
+
+class IDCardRenewer {
+  constructor() {
+    this.disposed = false;
+    this.redis = null;
+    this.refreshTimer = null;
+    this.nodeIdKey = null;
+    this.refreshDelay = 2000;
+
+    this.parentPort = parentPort;
+  }
+
+  async init(config) {
+    this.disposed = false;
+    this.nodeIdKey = config.nodeIdKey;
+    this.refreshDelay = config.refreshDelay;
+    
+    /**
+    * Since we do not have access to the Kuzzle Context we can't use kuzzle.ask('core:cache:internal:*',...),
+    * so we need to have an instance of Redis similar to the one used in the Cache Engine
+    */
+    await this.initRedis(config.redis);
+
+    /**
+     * Early renew, otherwise we need to wait the refreshDelay before the first ID Card renewal
+     * which could be enough to make the ID Card expire.
+     */
+    try {
+      await this.renewIDCard(); 
+    } catch (error) {
+      // Do nothing
```

Comment:
At least log something about the Redis error that could happend in the `renewCard` function